datastore/importers/nmdb-import-rpm/Parser.cpp

```diff
@@ -0,0 +1,101 @@
+// =============================================================================
+// Copyright 2022 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS). Under the terms of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+// =============================================================================
+// Maintained by Sandia National Laboratories <Netmeld@sandia.gov>
+// =============================================================================
+#include "Parser.hpp"
+
+// =============================================================================
+// Parser logic
+// =============================================================================
+Parser::Parser() : Parser::base_type(prestart)
+{
+  prestart = start [(qi::_val = pnx::bind(&Parser::getData, this))];
+
+  start =
+    *(packageLine [(pnx::bind(&Parser::addPackage, this, qi::_1))]
+      | ignoredLine
+      | qi::eol
+    )
+  ;
+
+  packageLine =
+    packageName [(pnx::bind(&nmdo::Package::setName, &qi::_val, qi::_1))]
+    > version [(pnx::bind(&nmdo::Package::setVersion, &qi::_val, qi::_1))]
+    > architecture [(pnx::bind(&nmdo::Package::setArchitecture, &qi::_val, qi::_1))]
+    > description [(pnx::bind(&nmdo::Package::setDescription, &qi::_val, qi::_1))]
+    > qi::eol
+  ;
+
+  packageName =
+    +token
+  ;
+  version =
+    +token
+  ;
+
+  architecture =
+    +token
+  ;
+
+  description =
+    +qi::ascii::print
+  ;
+
+  token =
+    +qi::ascii::graph
+  ;
```

Comment:
This isn't intuitive.

Ignoring the rule definition for a moment, by implementation the `token` matches one or more non-space characters, and `packageName`/`version`/`architecture`/`description` matches one or more `token`.  So, for example, the `packageName` rule should technically consume a whole line.  The only reason it doesn't is the rule definition which doesn't define a skipper.  So reading the rule implementation is counter intuitive on what it actually does.

Technically, there are a couple of ways to resolve this but the simplest is probably to remove the `+` from `token` in each of the rules.  Looking back, I think I missed this nugget in the `dpkg` version; we should probably fix that there as well.